manifests/install.pp

```diff
@@ -108,189 +111,118 @@
         command => 'touch ${package}_from_cache',
       }
     } else {
-      exec { "get_${package}_installer_${version}":
-        cwd     => $install_dir,
-        creates => "${install_dir}/${installerFilename}",
-        command => "wget -c --no-cookies --no-check-certificate --header \"Cookie: gpw_e24=http%3A%2F%2Fwww.oracle.com\" --header \"Cookie: oraclelicense=accept-securebackup-cookie\" \"${javaDownloadURI}\" -O ${installerFilename}",
-        timeout => 600,
-        require => Package['wget'],
-      }
-
-      file { "${install_dir}/${installerFilename}":
-        mode    => '0755',
-        require => Exec["get_${package}_installer_${version}"],
-      }
+      if ( $version in [ '7', '8' ] ) {
+        archive { "${install_dir}/${installer_filename}":
+          ensure        => present,
+          extract       => true,
+          source        => $java_download_uri,
+          proxy_server  => $jdk_oracle::proxy_host,
+          cookie        => 'oraclelicense=accept-securebackup-cookie',
+          extract_path  => "${java_home}/..",
+          extract_flags => '-xzof',
+          creates       => "${java_home}/bin/java",
+          cleanup       => true,
+          require       => File[$java_home],
+          user          => 'root',
+          group         => 'root'
+        }
+      } elsif ( $version == '6' ) {
+        exec { "get_${package}_installer_${version}":
+          cwd     => $install_dir,
+          creates => "${install_dir}/${installer_filename}",
+          command => "wget -c --no-cookies --no-check-certificate --header \"Cookie: gpw_e24=http%3A%2F%2Fwww.oracle.com\" --header \"Cookie: oraclelicense=accept-securebackup-cookie\" \"${java_download_uri}\" -O ${installer_filename}",
+          timeout => 600,
+          require => Package['wget'],
+        }
+        exec { "extract_${package}_${version}":
+          cwd     => "${install_dir}/",
+          command => "${install_dir}/${installer_filename}",
+          creates => $java_home,
+          require => File["${install_dir}/${installer_filename}"],
+        }
+        file { "${install_dir}/${installer_filename}":
+          mode    => '0755',
+          require => Exec["get_${package}_installer_${version}"],
+        }
 
-      if ! defined(Package['wget']) {
-        package { 'wget':
-          ensure =>  present,
+        if ! defined(Package['wget']) {
```

Comment:
Humm, i think, since i'm not using wget that i've not been focused on this particular package.
Noticing your remark, create_resources is used to iterate over hashes. eg. define a whole subset to resource entries, as ensure_resources ensure that the specified resources are defined, not raising resource duplicated.

But you are true with the way i've defined packages in init.pp, it should have been using ```ensure_packages(['curl','unzip'], {'ensure' => 'present'})``` for clarity and coding style.

I'll push updates for the packages.